docker/Dockerfile-carla-mot

```diff
@@ -0,0 +1,35 @@
+ARG base_image_tag
+
+FROM twosixarmory/pytorch:${base_image_tag}
+
+RUN pip install --no-cache-dir \
+    git+https://github.com/ifzhang/ByteTrack.git \
+    thop \
+    lap \
+    Cython
+
+# Requires cython for install, so will fail if run in the same pip install as cython
+RUN pip install --no-cache-dir \
+    cython-bbox
+
+
+WORKDIR /armory-repo
+
+RUN echo "Updating pip" && \
+    pip install --upgrade pip && \
+    echo "Building ART" && \
+    pip install --no-cache-dir \
+    adversarial-robustness-toolbox==1.12.1
+
+# NOTE: This COPY command is filtered using the `.dockerignore` file
+#       in the root of the repo.
+COPY ./ /armory-repo
+
+RUN echo "Building Armory from local source" && \
+    pip install --no-compile --no-cache-dir --editable . && \
+    echo "Configuring Armory..." && \
+    armory configure --use-default && \
+    echo "Cleaning up..." && \
+    rm -rf /armory-repo/.git
```

Comment:
We're building from `pytorch`, not `base`, so these lines should be unnecessary.

Reply:
(lines 16-33)

Reply:
They are necessary until the updated images are uploaded into dockerhub.

Reply:
See commits [6e17af8](https://github.com/twosixlabs/armory/pull/1663/commits/6e17af8f66787ee787e60ada28efadb90ca32f83) [c8cff4c](https://github.com/twosixlabs/armory/pull/1663/commits/c8cff4c3dfbb46de2810ca120f4b580ab96f3bf7) [9d16176](https://github.com/twosixlabs/armory/pull/1663/commits/9d161766b04b38ad62e2e8e11eb05fd46d4b4686)

Reply:
A clarification for my sake: because the carla-mot dockerfile starts with `FROM twosixlabs/pytorch` the CI build gets the latest `pytorch` which has version 0.15.4. This apparently redundant "Building Armory from local source" ensures that code and version are refreshed.

If you run `docker build pytorch` followed by `docker build carla-mot` on your local, *stateful* machine, this step is not needed. Because the CI including release is *stateless* between runs this is needed.  There was a [CI failure](https://github.com/twosixlabs/armory/actions/runs/3221627521/jobs/5270015780) earlier which shows the fault.

Reply:
Flagged for later build time optimization in #1671

Reply:
Hmmm. Maybe we should build from the `base` image, then, instead of `pytorch`.